internal/daemon/controller/handlers/targets/target_service.go

```diff
@@ -268,37 +268,82 @@ func (s Service) ListTargets(ctx context.Context, req *pbs.ListTargetsRequest) (
 	}
 
 	var listResp *pagination.ListResponse[target.Target]
-	if req.GetRefreshToken() == "" {
+	sortBy := "created_time"
+	if req.GetListToken() == "" {
 		listResp, err = target.List(ctx, grantsHash, pageSize, filterItemFn, repo)
 		if err != nil {
 			return nil, err
 		}
 	} else {
-		rt, err := handlers.ParseRefreshToken(ctx, req.GetRefreshToken())
+		pt, err := handlers.ParseListToken(ctx, req.GetListToken())
 		if err != nil {
 			return nil, err
 		}
-		// We're doing the conversion from the protobuf types to the
-		// domain types here rather than in the domain so that the domain
-		// doesn't need to know about the protobuf types.
-		domainRefreshToken, err := refreshtoken.New(
-			ctx,
-			rt.CreatedTime.AsTime(),
-			rt.UpdatedTime.AsTime(),
-			handlers.RefreshTokenResourceToResource(rt.ResourceType),
-			rt.GrantsHash,
-			rt.LastItemId,
-			rt.LastItemUpdatedTime.AsTime(),
-		)
-		if err != nil {
-			return nil, err
-		}
-		if err := domainRefreshToken.Validate(ctx, resource.Target, grantsHash); err != nil {
-			return nil, err
-		}
-		listResp, err = target.ListRefresh(ctx, grantsHash, pageSize, filterItemFn, domainRefreshToken, repo)
-		if err != nil {
-			return nil, err
+		switch st := pt.Token.(type) {
```

Comment:
When we have a list token, we have 3 different scenarios:
1. We need to build a second page for the initial pagination phase. This is using a `ListToken_PaginationToken`.
2. We need to start a refresh phase. This is done using a `ListToken_StartRefreshToken`.
3. We need to continue a refresh phase. This is done using a `ListToken_RefreshToken`.